src/OSPSuite.Infrastructure.Export/ExportToExcelTask.cs

```diff
@@ -29,24 +31,40 @@ public static void ExportDataTableToExcel(DataTable dataTable, string fileName,
       /// <param name="fileName">Target file</param>
       /// <param name="openExcel">If set to true, excel will be launched with the exported file</param>
       /// <param name="workBookConfiguration"></param>
-      public static void ExportDataTablesToExcel(IEnumerable<DataTable> dataTables, string fileName, bool openExcel, IWorkbookConfiguration workBookConfiguration = null)
+      public static void ExportDataTablesToExcel(IReadOnlyList<DataTable> dataTables, string fileName, bool openExcel, IWorkbookConfiguration workBookConfiguration = null)
       {
-         var tables = dataTables.ToList();
-         if (workBookConfiguration == null)
-            workBookConfiguration = new WorkbookConfiguration();
+         var workBookConfigurationToUse = workBookConfiguration ?? new WorkbookConfiguration();
+         updateTableNamesToMaxAllowedLengthAndUnicity(dataTables);
 
-         for (var i = 0; i < tables.Count(); i++)
-         {
-            var dataTable = tables.ElementAt(i);
-            exportDataTableToWorkBook(workBookConfiguration, dataTable);
-         }
+         dataTables.Each(x => exportDataTableToWorkBook(x, workBookConfigurationToUse));
 
-         saveWorkbook(fileName, workBookConfiguration.WorkBook);
+         saveWorkbook(fileName, workBookConfigurationToUse.WorkBook);
 
          if (openExcel)
             FileHelper.TryOpenFile(fileName);
       }
 
+      private static void updateTableNamesToMaxAllowedLengthAndUnicity(IReadOnlyList<DataTable> dataTables)
+      {
+         //31 is max allowed by excel, we remove some more to allow for uniqueness (we assume we won't have more than 100 number and one extra space for the separator => 4)
+         const int maxTableNameLength = 31 - 4;
```

Comment:
All right so there is a MAX length for an excel sheet name 
This is the reason why it crashes. So I need to make sure ALL names are smaller than the max and also add some extra buffer to make sure they are also unique